### ISSUE TITLE
"Baseline" in feature_contrib columns

### ISSUE THREAD

**Issue by tuckergordon** [Microsoft/LightGBM#1448]
In the `NumPredictOneRow(..)` function in gbdt.h, there is an extra column given for "baseline":

`num_preb_in_one_row = num_tree_per_iteration_ * (max_feature_idx_ + 2); // +1 for 0-based indexing, +1 for baseline`

It is not clear what this "baseline" is. When run, this produces the same value for every row. If I am missing an explanation in the code somewhere, please let me know. Otherwise, it might be a good idea to add a bit more documentation.

**Comment by guolinke** [comment]
refer to paper: https://github.com/slundberg/shap

and ping @slundberg

**Comment by slundberg** [comment]
Yes, it's the average over the training dataset. Where the average is taken before any margin transform is taken.
